src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -20228,13 +20228,24 @@ void Unit::OutDebugInfo() const
 class AuraMunchingQueue : public BasicEvent
 {
 public:
-    AuraMunchingQueue(Unit& owner, ObjectGuid targetGUID, int32 basePoints, uint32 spellId) : _owner(owner), _targetGUID(targetGUID), _basePoints(basePoints), _spellId(spellId) { }
+    AuraMunchingQueue(Unit& owner, ObjectGuid targetGUID, int32 basePoints, uint32 spellId, AuraEffect* aurEff, AuraType auraType) : _owner(owner), _targetGUID(targetGUID), _basePoints(basePoints), _spellId(spellId), _aurEff(aurEff), _auraType(auraType) { }
 
     bool Execute(uint64 /*eventTime*/, uint32 /*updateTime*/) override
     {
         if (_owner.IsInWorld() && _owner.FindMap())
+        {
+            bool auraFound = false;
+            Unit::AuraEffectList const& auraEffects = _owner.GetAuraEffectsByType(_auraType);
```

Comment:
I think I mixed up the owner and the target. This aura belongs to the target (right?), so we need to check the aura on the target, not the owner.

Reply:
You seem to be correct, adjusted the PR accordingly.